### PR TITLE
Reformat Shuffle

### PR DESCRIPTION
**Pull request by rimshasaeed**
Shuffle #5513

### CODE REVIEW

**Comment by rimshasaeed** [comment]
@abdrahmandiab could you please indicate where the tests are going wrong? And how to run these tests manually?

File: ivy/container/random.py [hunk 1]
```diff
@@ -1090,6 +1108,24 @@ def shuffle(
...
+
+        >>> x = ivy.Container(a=ivy.array([5,2,9]), \
+                              b=ivy.array([7,1,6]))
+        >>> x.shuffle()
+        {
+            a: ivy.array([9, 5, 2]),
+            b: ivy.array([6, 7, 1])
+        }
+
+        With :class:`ivy.Array` inputs:
```
Comment (abdrahmandiab) [inline ivy/container/random.py:1124]: this example with `ivy.Array` only is not relevant to the Container instance method and should be removed 😅

File: ivy/functional/ivy/random.py [hunk 2]
```diff
@@ -536,12 +536,17 @@ def shuffle(
     ret
         An array object, shuffled along the first dimension.
 
-    Examples
-    --------
+    Functional Examples
```
Comment (abdrahmandiab) [inline ivy/functional/ivy/random.py:539]: Please revert this back to just `Examples`, the `Functional Examples` 😅

File: ivy/functional/ivy/random.py [hunk 3]
```diff
@@ -536,12 +536,17 @@ def shuffle(
...
+    -------------------
     >>> x = ivy.array([1, 2, 3, 4, 5])
     >>> y = ivy.shuffle(x)
     >>> print(y)
     ivy.array([2, 1, 4, 3, 5])
 
+    >>> x = ivy.array([1,0,5])
+    >>> y = ivy.array(ivy.zeros((3,)))
```
Comment (abdrahmandiab) [inline ivy/functional/ivy/random.py:547]: please use change this to `ivy.array([ 0, 0, 0 ])` as it is more clear than using a function call 😅

File: ivy/functional/ivy/random.py [hunk 4]
```diff
@@ -536,12 +536,17 @@ def shuffle(
...
+    >>> z = ivy.shuffle(x, out=y)
+    >>> print(y)
+    ivy.array([0, 5, 1])
     """
```
Comment (abdrahmandiab) [inline ivy/functional/ivy/random.py:551]: An extra example should be added here showing an example of the function call with ivy.Container as input

**Review by abdrahmandiab — CHANGES_REQUESTED** [review]
Hi @rimshasaeed , 
Thank you so much for your hard work so far! I have added some review comments pointing out some edits that need to be made, and it should all be very easy hopefully!

Additionally, could you please add the reformatting checklist to your PR by following the instructions [here](https://lets-unify.ai/ivy/contributing/4_open_tasks.html#formatting-checklist) and then check it off according to the `Legend` section? 

It would make the review process a lot easier and quicker, and would ensure we don't miss any cases as well! 😃 

Thank you so much, and please keep going! 🚀

**Comment by rimshasaeed** [comment]
add_reformatting_checklist_reformatting

**Comment by rimshasaeed** [comment]
add_reformatting_checklist_random

File: ivy/container/random.py [hunk 5]
```diff
@@ -1039,6 +1039,24 @@ def static_shuffle(
         -------
         ret
             A container object, shuffled along the first dimension.
+
+        Examples
+        --------
+        >>> x = ivy.Container(a=ivy.array([7,6,0]), \
```
Comment (abdrahmandiab) [inline ivy/container/random.py:1045]: Hi, it seems you missed changing the `\` notation in all files as I instructed in my previous review, could you please change these as well? 😅

File: ivy/container/random.py [hunk 6]
```diff
@@ -1039,6 +1039,24 @@ def static_shuffle(
...
+                              b=ivy.array([8,9,4]))
+        >>> ivy.Container.static_shuffle(x)
+        {
+            a: ivy.array([7, 0, 6]),
+            b: ivy.array([8, 4, 9])
+        }
+
+        >>> x = ivy.Container(a=ivy.array([0,0,0]), \
```
Comment (abdrahmandiab) [inline ivy/container/random.py:1053]: This example with `0`s being shuffled is not interesting and should be removed as it does not display any of the function's behavior. The above example alone is sufficient.

File: ivy/functional/ivy/random.py [hunk 7]
```diff
@@ -538,10 +538,39 @@ def shuffle(
...
+    {
+        a: ivy.array([5, 9, 2]),
+        b: ivy.array([6, 1, 7])
+    }
+
+    >>> x = ivy.Container(a=ivy.array([8,2,5]),
+    ...                   b=ivy.array([3,9,0]))
+    >>> y = ivy.Container(a=ivy.array([0,0,0]),
+    ...                   b=ivy.array([0,0,0]))
+    >>> z = ivy.shuffle(x, out=y)
```
Comment (abdrahmandiab) [inline ivy/functional/ivy/random.py:569]: Could you instead change this example to be
```
>>> x = ivy.Container( a=ivy.array([8,2,5]),
...                    b=ivy.array([3,9,0]))
>>> ivy.shuffle( x, out=x)
>>> print(x)
# then the output here
```
To show an example of the `out` argument being used to inplace update `x` as described in point *6.d* of the checklist.

File: ivy/functional/ivy/random.py [hunk 8]
```diff
@@ -538,10 +538,39 @@ def shuffle(
...
+    With :class:`ivy.Array` input:
+
     >>> x = ivy.array([1, 2, 3, 4, 5])
     >>> y = ivy.shuffle(x)
     >>> print(y)
     ivy.array([2, 1, 4, 3, 5])
 
+    >>> x = ivy.array([1,0,5])
+    >>> y = ivy.array([0, 0, 0])
+    >>> z = ivy.shuffle(x, out=y)
```
Comment (abdrahmandiab) [inline ivy/functional/ivy/random.py:550]: no need to put `z = ` here, the line can just be `>>> ivy.shuffle(x, out=y)`

File: ivy/array/random.py [hunk 9]
```diff
@@ -372,5 +372,25 @@ def shuffle(
         -------
         ret
             An array object, shuffled along the first dimension.
+
+        Examples
+        --------
+        >>> x = ivy.Container(a=ivy.array([5,2,9]),
```
Comment (abdrahmandiab) [inline ivy/array/random.py:378]: The examples here are incorrect unfortunately. 😅 
Please refer to the [Docstring Examples](https://lets-unify.ai/ivy/deep_dive/docstring_examples.html#docstring-examples) section of the docs, or see point `6.j` of the checklist for correct instruction on what the example for the `array` class should be like.

File: ivy/functional/ivy/random.py (hunk 8 above)
Comment (abdrahmandiab) [reply ivy/functional/ivy/random.py:550]: Additionally, could add a value for the `seed` parameter here as it is required to show each parameter being used at least once

**Review by abdrahmandiab — CHANGES_REQUESTED** [review]
Hi @rimshasaeed , 

I've gone ahead and updated the checklist according to the work you've committed so far, so please take note of the points that have not been checked.

Additionally, I've left some comments outlining some errors in your implementation that should be addressed.

Thanks! 😄

File: ivy/container/random.py (hunk 6 above)
Comment (rimshasaeed) [reply ivy/container/random.py]: Done, removed this example with `0`s! 😅

File: ivy/functional/ivy/random.py (hunk 7 above)
Comment (rimshasaeed) [reply ivy/functional/ivy/random.py]: I have added two examples here for `out` argument. One uses `out` to update a new array `y`, and the other one uses `out` to inplace update the input array

File: ivy/functional/ivy/random.py (hunk 8 above)
Comment (rimshasaeed) [reply ivy/functional/ivy/random.py]: Done, fixed this. Also, added a separate example for `seed` argument.

File: ivy/array/random.py (hunk 9 above)
Comment (rimshasaeed) [reply ivy/array/random.py]: Done, reformatted these examples to Array Instance Method

**Review by abdrahmandiab — APPROVED** [review]
Hi @rimshasaeed,
incredibly sorry you had to wait so long for a review! 😅 I've just been quite busy lately!
I have fixed a tiny mistake with your array instance method docstring example implementation and the PR is ready to merge now!

Thank you so much for your work!

**Comment by rimshasaeed** [comment]
Thank you very much for reviewing my PR!

**Pull request merged**